### ISSUE TITLE
[a11y]: BigNumbers: stop using random H4 heading

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#7167]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.62

### React version

18

### Automated testing tool and ruleset

accessibility-checker

### Description

As explained in https://www.w3.org/WAI/tutorials/page-structure/headings/

> Skipping heading ranks can be confusing and should be avoided where possible: Make sure that a `<h2>` is not followed directly by an `<h4>`, for example. 

So, `BigNumber` shouldn't randomly use an `<h4>` heading.

But further, it probably shouldn't use a heading at all.  It seems like overkill (and annoying for blind users) if every number has its own heading.

Maybe `<dl>`, `<dt>`, and `<dd>` makes more sense.

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/experimental-components-big-numbers-bignumbers--big-numbers&globals=viewport:basic

### Steps to reproduce

Chrome developer tools.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by wkeese** [comment]
OK thanks.  FYI, https://stackoverflow.com/questions/33214093/is-it-good-practice-to-use-label-for-non-input-non-interactive-elements lists some other options of using `<label>` or just `<span>`.   I wonder if IBM's accessibility-checker would complain if you used `<label>`.

**Comment by elycheea** [comment]
Thanks @wkeese for the insight. We’re leaning towards `span` as the safer option.

### Acceptance criteria

- [ ] Update `BigNumbers` to use `span` instead of `h4`
